store.go

```diff
@@ -562,6 +562,15 @@ type LayerOptions struct {
 	UncompressedDigest digest.Digest
 	// True is the layer info can be treated as volatile
 	Volatile bool
+	// BigData is a set of items which should be stored with the layer.
+	BigData []LayerBigDataOption
+	// Flags is a set of named flags and their values to store with the layer.
+	Flags map[string]interface{}
```

Comment:
This doesn't go as far as making them generally editable, just settable at the time the record is created.  There have been points where we've found ourselves wanting to be able to attach bits of information to records of one type or another, where coordinating them in the one general `Metadata` field would have been too much overhead, and this could have been enough.

Adding notes to the various options fields to note that flags can only be set at create-time, but that could change at some point.

Reply:
Documenting the flag limitations LGTM.